express/src/middleware/mobileValidator.ts

```diff
@@ -1,16 +1,80 @@
 import {NextFunction, Request, Response} from "express";
 
 export function mobileValidator(req: Request, res: Response, next: NextFunction) {
-    // remove whitespace
-    // ensure UK mobile
-    // convert to internation format
+    let mobileNumber = req.body.mobileNumber;
+    mobileNumber = removeSpaceAndDash(mobileNumber);
 
-    // modify req.body.telephone
-    let errorMessages = new Map<string, string>();
-    let values = new Map<string, string>();
-    if(true) {
-        next();
-    } else {
-        res.render('/create-account/get-email.njk', {fieldOrder: ['telephone'], values: values, errorMessages: errorMessages})
+    if ( mobileNumber === "" || mobileNumber === null ){
+        errorResponse(mobileNumber, res , 'mobileNumber', 'Mobile number is required');
+        return;
+    }
+    
+    if (!onlyNumbers(mobileNumber)){
+        errorResponse(mobileNumber, res , 'mobileNumber', 'Make sure there are no letters in the mobile number.');
+        return;
+    }
+
+    if(!ukNumberCheck(mobileNumber)){
+        errorResponse(mobileNumber, res , 'mobileNumber', 'Make sure you have enterd a correct UK phone number .');
+        return;
+    }
+    if (beginsWith070(mobileNumber)) {
+        errorResponse(mobileNumber, res , 'mobileNumber', 'We do not accept premium rate telephone numbers.');
+        return;
     }
-}
\ No newline at end of file
+
+    mobileNumber = convert07Numbers(mobileNumber);
+
+    if (!lengthCheck(mobileNumber)) {
+        errorResponse(mobileNumber, res , 'mobileNumber', 'Mobile number must be 10 digits long.');
```

Comment:
It would be better to just say the number contains the incorrect number of digits instead of asking them to take off the 0 or +44 and then count the number of digits.